modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -69,6 +71,68 @@ def wait(*args, **kwargs):
     return func
 
 
+def progress_bar_wrapper(f):
+    """
+    Wrap computation function inside a progress bar.
+
+    Spawns another thread which displays a progress bar showing
+    estimated completion time.
+
+    Parameters
+    ----------
+    f : callable
+        The name of the function to be wrapped.
+
+    Returns
+    -------
+    callable
+        Decorated version of `f` which reports progress.
+    """
+    if Engine.get() != "Ray":
+        if ProgressBar.get():
+            raise NotImplementedError(
+                f"ProgressBar doesn't work with '{Engine.get()}' engine."
+            )
+        else:
+            return f
+
+    from functools import wraps
+    from modin.core.execution.ray.generic.modin_aqp import call_progress_bar
```

Comment:
This progress bar increment is specific to the ray engine, are you sure this works?

Reply:
Please note, that progress bar is enabled only for ray engine case.

That is, it has a specific implementation (at the moment), but its usage can always be the same, even when implementations for other engines appear. Therefore, the use of the wrapper itself can be moved to the base class.

Reply:
I don't like we import something from ray directory to pandas, which is base for ray. We should probably have some base structure for the progress bar, which requires us do more complex refactor.

Reply:
You are talking about the solution to which you need to strive, I agree. However, it is much more difficult to implement because we need to know the implementation details of the progress bar. This decision is the most balanced in relation to resources/result. I assume that the current version brings more benefits than problems.